src/build/__main__.py

```diff
@@ -40,15 +40,19 @@ def _error(msg, code=1):  # type: (str, int) -> None  # pragma: no cover
     exit(code)
 
 
-def _build_in_isolated_env(builder, outdir, distributions):  # type: (ProjectBuilder, str, List[str]) -> None
+def _build_in_isolated_env(srcdir, config_settings, outdir, distributions):
+    # type: (str, Optional[ConfigSettings], str, List[str]) -> None
     with IsolatedEnvironment.for_current() as env:
+        builder = ProjectBuilder(srcdir, config_settings, python_executable=env.executable)
```

Comment:
I'm not a fan of duplicating the project builder creation and moving it inside the respective methods. I prefer my solution from https://github.com/gaborbernat/python-build/blob/50c6a93891/src/build/__main__.py#L44

Reply:
Hum, sure, we can do that.

Reply:
I have made `python_executable` as property that mirrors the hook's `python_executable`. I think we should make the hook private, that can come in the next PR.